packaging: Fix snapshot packaging

cathay4t opened the pull request:
With rust-srpm-macros-17-4.el9, `%cargo_pre` without `-V` option will
put the crates.io registry to empty folder, hence break all snapshot rpm
builds.

The fix is use create vendor tarball for snapshot rpm build.

Also fixed Fedora compiling issue by removing the ping of ctrlc crate
version.

ffmancera commented:
packit seems to be broken now. I think it is related to this change considering it didn't fail before.

cathay4t commented:
Packit fixed

cathay4t committed:
packaging: Fix snapshot packaging

With rust-srpm-macros-17-4.el9, `%cargo_pre` without `-V` option will
put the crates.io registry to empty folder, hence break all snapshot rpm
builds.

The fix is use create vendor tarball for snapshot rpm build.

Also fixed Fedora compiling issue by removing the pin of ctrlc crate
version.

Signed-off-by: Gris Ge <fge@redhat.com>
